ui/plugin-system/src/model/variables.ts

```diff
@@ -18,29 +18,30 @@ import { usePlugin } from '../components/PluginLoadingBoundary';
  * Plugin for handling custom VariableDefinitions.
  */
 export interface VariablePlugin<Options extends JsonObject = JsonObject> {
-  useVariableOptions: UseVariableOptionsHook<Options>;
+  getVariableOptions: GetVariableOptions<Options>;
 }
 
 /**
  * Plugin hook responsible for getting the options of a custom variable
  * definition.
  */
-export type UseVariableOptionsHook<Options extends JsonObject = JsonObject> = (
+export type GetVariableOptions<Options extends JsonObject = JsonObject> = (
   definition: ListVariableDefinition<Options>
```

Comment:
I'm wondering if this should receive the full ListVariable definition or just the Options for the variable kind?